From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Thu, 14 Dec 2023 14:36:46 +0100
Subject: [PATCH] arm64: dts: i.MX8MP: skov-reva: set PMIC restart handler
 priority to 200

The normal reset on the i.MX8MP happens via the i.MX watchdog.
After reset, the watchdog's WSTR register can be consulted to find out
whether reboot was intentional or because the watchdog timeout expired.

This scheme doesn't work as expected when a watchdog restart causes the
PMIC to power cycle the CPU rail: In that case, both types of reset would
be detected as power-on reset if the watchdog is asked or a watchdog
reset if the PMIC is asked.

Fix this by using the PMIC reset for the regular reboot. That way we can
ask the PMIC after a reset what the reset reason is and will get an
accurate answer without having to sacrifice cold resets.

Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
---
 arch/arm64/boot/dts/freescale/imx8mp-skov-reva.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-skov-reva.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-skov-reva.dtsi
index b5e34e9d5a90..1a6e11cef391 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-skov-reva.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp-skov-reva.dtsi
@@ -252,6 +252,7 @@ pmic@25 {
 		pinctrl-0 = <&pinctrl_pmic>;
 		interrupts-extended = <&gpio1 3 IRQ_TYPE_EDGE_RISING>;
 		sd-vsel-gpios = <&gpio1 4 GPIO_ACTIVE_HIGH>;
+		priority = <200>;
 
 		regulators {
 			reg_vdd_soc: BUCK1 {
